modules/mediagoBidAdapter.js

```diff
@@ -0,0 +1,364 @@
+    ].join(''), 'i');
+    if (reg1.test(a) ||
+      reg2.test(a.substr(0, 4))) {
+      check = true;
+    }
+  })(navigator.userAgent || navigator.vendor || window.opera);
+  return check;
+}
+
+/**
+ * 将尺寸转为RTB识别的尺寸
+ *
+ * @param  {Array|Object} requestSizes 配置尺寸
+ * @return {Object}
+ */
+function transformSizes(requestSizes) {
+  let sizes = [];
+  let sizeObj = {};
+
+  if (utils.isArray(requestSizes) && requestSizes.length === 2 &&
+    !utils.isArray(requestSizes[0])) {
+    sizeObj.width = parseInt(requestSizes[0], 10);
+    sizeObj.height = parseInt(requestSizes[1], 10);
+    sizes.push(sizeObj);
+  } else if (typeof requestSizes === 'object') {
+    for (let i = 0; i < requestSizes.length; i++) {
+      let size = requestSizes[i];
+      sizeObj = {};
+      sizeObj.width = parseInt(size[0], 10);
+      sizeObj.height = parseInt(size[1], 10);
+      sizes.push(sizeObj);
+    }
+  }
+
+  return sizes;
+}
+
+/**
+ * 获取广告位配置
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getItems(validBidRequests, bidderRequest) {
+  let items = [];
+  items = validBidRequests.map((req, i) => {
+    let ret = {};
+    let mediaTypes = getProperty(req, 'mediaTypes');
+
+    let sizes = transformSizes(getProperty(req, 'sizes'));
+    let matchSize;
+
+    // 确认尺寸是否符合我们要求
+    for (let size of sizes) {
+      if (size.width === 300 && size.height === 250) {
+        matchSize = size;
+        break;
+      }
+    }
+
+    // if (mediaTypes.native) {}
+    // banner广告类型
+    if (mediaTypes.banner) {
+      let id = '' + (i + 1);
+      ret = {
+        id: id,
+        // bidFloor: 0, // todo
+        banner: {
+          h: matchSize.height,
+          w: matchSize.width,
+          pos: 1,
+        }
+      };
+      itemMaps[id] = {
+        req,
+        ret
+      };
+    }
+
+    return ret;
+  });
+  return items;
+}
+
+/**
+ * 获取rtb请求参数
+ *
+ * @param {Array}  validBidRequests an an array of bids
+ * @param {Object} bidderRequest  The master bidRequest object
+ * @return {Object}
+ */
+function getParam(validBidRequests, bidderRequest) {
+  // console.log(validBidRequests, bidderRequest);
+  let isMobile = isMobileAndTablet() ? 1 : 0;
+  let isTest = 0;
+  let auctionId = getProperty(bidderRequest, 'auctionId') || getRandomId();
+  let items = getItems(validBidRequests, bidderRequest);
+
+  const domain = document.domain;
+  const location = utils.deepAccess(bidderRequest, 'refererInfo.referer');
+
+  if (items && items.length) {
+    let c = {
+      'id': 'mgprebidjs_' + auctionId,
+      'test': +isTest,
+      'at': 1,
+      'cur': ['USD'],
+      'device': {
+        // 'dnt':0,
+        // 'devicetype':2,
+        'js': 1,
+        'os': navigator.platform || '',
+        'ua': navigator.userAgent,
+        'language': /en/.test(navigator.language) ? 'en' : navigator.language,
+        'ip': '64.124.253.1', // to delete
```

Comment:
what's happening here

Reply:
Sorry, our demand is based on US area. So I test it with US IP. I have deleted it.